plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -89,29 +113,45 @@ private static boolean hasUserOptedIn() {
    * opting out. (Therefore, this method essentially returns true if the opt-in dialog was ever
    * presented before.)
    */
-  private static boolean hasUserRegisteredOptInStatus() {
-    return getBooleanPreference(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, false);
+  private boolean hasUserRegisteredOptInStatus() {
+    return preferences.getBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, false);
   }
 
-  static void registerOptInStatus(boolean optedIn) {
-    IEclipsePreferences prefs = ConfigurationScope.INSTANCE.getNode(PREFERENCES_PLUGIN_ID);
-    if (prefs != null) {
-      prefs.putBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN, optedIn);
-      prefs.putBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, true);
-      flushPreferences(prefs);
-    }
+  void registerOptInStatus(boolean optedIn) {
```

Comment:
My concern is that OPT_IN_REGISTERED is not a preference at all. I.e. the user does not set this.